src/skmatter/decomposition/_pcovc.py

```diff
@@ -0,0 +1,425 @@
+        self.pxz_ = self.pxt_ @ self.ptz_
+
+        if len(Y.shape) == 1 and type_of_target(Y) == "binary":
+            self.pxz_ = self.pxz_.reshape(
+                X.shape[1],
+            )
+            self.ptz_ = self.ptz_.reshape(
+                self.n_components_,
+            )
+
+        self.components_ = self.pxt_.T  # for sklearn compatibility
+        return self
+
+    def _fit_feature_space(self, X, Y, Z):
+        r"""In feature-space PCovC, the projectors are determined by:
+
+        .. math::
+            \mathbf{\tilde{C}} = \alpha \mathbf{X}^T \mathbf{X} +
+            (1 - \alpha) \left(\left(\mathbf{X}^T
+            \mathbf{X}\right)^{-\frac{1}{2}} \mathbf{X}^T
+            \mathbf{Z}\mathbf{Z}^T \mathbf{X} \left(\mathbf{X}^T
+            \mathbf{X}\right)^{-\frac{1}{2}}\right)
+
+        where
+
+        .. math::
+            \mathbf{P}_{XT} = (\mathbf{X}^T \mathbf{X})^{-\frac{1}{2}}
+                                \mathbf{U}_\mathbf{\tilde{C}}^T
+                                \mathbf{\Lambda}_\mathbf{\tilde{C}}^{\frac{1}{2}}
+
+        .. math::
+            \mathbf{P}_{TX} = \mathbf{\Lambda}_\mathbf{\tilde{C}}^{-\frac{1}{2}}
+                                \mathbf{U}_\mathbf{\tilde{C}}^T
+                                (\mathbf{X}^T \mathbf{X})^{\frac{1}{2}}
+        """
+        return super()._fit_feature_space(X, Y, Z, compute_pty_=False)
+
+    def _fit_sample_space(self, X, Y, Z, W):
+        r"""In sample-space PCovC, the projectors are determined by:
+
+        .. math::
+            \mathbf{\tilde{K}} = \alpha \mathbf{X} \mathbf{X}^T +
+            (1 - \alpha) \mathbf{Z}\mathbf{Z}^T
+
+        where
+
+        .. math::
+            \mathbf{P}_{XT} = \left(\alpha \mathbf{X}^T + (1 - \alpha)
+                               \mathbf{W} \mathbf{Z}^T\right)
+                               \mathbf{U}_\mathbf{\tilde{K}}
+                               \mathbf{\Lambda}_\mathbf{\tilde{K}}^{-\frac{1}{2}}
+
+        .. math::
+            \mathbf{P}_{TX} = \mathbf{\Lambda}_\mathbf{\tilde{K}}^{-\frac{1}{2}}
+                                \mathbf{U}_\mathbf{\tilde{K}}^T \mathbf{X}
+        """
+        return super()._fit_sample_space(X, Y, Z, W, compute_pty_=False)
+
+    def inverse_transform(self, T):
+        r"""Transform data back to its original space.
+
+        .. math::
+            \mathbf{\hat{X}} = \mathbf{T} \mathbf{P}_{TX}
+                              = \mathbf{X} \mathbf{P}_{XT} \mathbf{P}_{TX}
+
+        Parameters
+        ----------
+        T : ndarray, shape (n_samples, n_components)
+            Projected data, where n_samples is the number of samples
+            and n_components is the number of components.
+
+        Returns
+        -------
+        X_original ndarray, shape (n_samples, n_features)
+        """
+        return super().inverse_transform(T)
+
+    def decision_function(self, X=None, T=None):
+        """Predicts confidence scores from X or T."""
```

Comment:
Should touch this up further. Here's a start
```suggestion
r"""Predicts confidence scores from X or T.
.. math::
    \mathbf{Z} = \mathbf{T} \mathbf{P}_{TZ}
                      = \mathbf{X} \mathbf{P}_{XT} \mathbf{P}_{TZ}

        Parameters
        ----------
        X : ndarray, shape(n_samples, n_features)
             Original data for which we want to get confidence scores, where n_samples is the number of samples and n_features is the number of features.

        T : ndarray, shape (n_samples, n_components)
            Projected data for which we want to get confidence scores, where n_samples is the number of samples
            and n_components is the number of components.
        
        Returns
        --------
        Z
```